### ISSUE TITLE
Unable to Open Model

### ISSUE THREAD

**Issue by HeroponRikiBestest** [rebane2001/matterport-dl#137]
### One or more sample matterport IDs / urls that reproduce the issue

9MxxdwzN2GQ

### Started up message

Started up a download run Running python 3.12.3 on linux with matterport-dl version: refs/heads/main (d65735e8f19016b25c97637e7a842374204562c4)

### Steps to reproduce

Try and open the model locally via matterport after downloading.

### What went wrong

The resulting page shows "Oops, model not available.". The download itself had no errors. 
`Done, Total fetches: 70370 Skipped: 0 (0%) actual Request: 70370 (100%) Success: 70097 (100%) Failed403: 273 (0%) Failed404: 0 (0%) FailedUnknown: 0 (0%)!`

The error logs in browser when trying to run it locally:

```

[model-api-client] 0.301s Error: Server response was missing for query 'GetModelPolicies'.
    t http://127.0.0.1:8080/js/showcase.js:2
    ct http://127.0.0.1:8080/js/showcase.js:2
    c http://127.0.0.1:8080/js/showcase.js:2
    c http://127.0.0.1:8080/js/showcase.js:2
    y http://127.0.0.1:8080/js/showcase.js:17
    b http://127.0.0.1:8080/js/showcase.js:17
    value http://127.0.0.1:8080/js/showcase.js:17
    ie http://127.0.0.1:8080/js/showcase.js:2
    ie http://127.0.0.1:8080/js/showcase.js:2
    error http://127.0.0.1:8080/js/showcase.js:2
    y http://127.0.0.1:8080/js/showcase.js:17
    b http://127.0.0.1:8080/js/showcase.js:17
    value http://127.0.0.1:8080/js/showcase.js:17
    y http://127.0.0.1:8080/js/showcase.js:17
    b http://127.0.0.1:8080/js/showcase.js:17
    value http://127.0.0.1:8080/js/showcase.js:17
    p http://127.0.0.1:8080/js/showcase.js:2
    promise callback*97704/p/</< http://127.0.0.1:8080/js/showcase.js:2
    e http://127.0.0.1:8080/js/showcase.js:17
    value http://127.0.0.1:8080/js/showcase.js:17
    l http://127.0.0.1:8080/js/showcase.js:2
    promise callback*41598/f/l</< http://127.0.0.1:8080/js/showcase.js:2
    e http://127.0.0.1:8080/js/showcase.js:17
    value http://127.0.0.1:8080/js/showcase.js:17
    complete http://127.0.0.1:8080/js/showcase.js:2
    start http://127.0.0.1:8080/js/showcase.js:2
    t http://127.0.0.1:8080/js/showcase.js:2
    getObservableFromLink http://127.0.0.1:8080/js/showcase.js:2
    getResultsFromLink http://127.0.0.1:8080/js/showcase.js:2
    m http://127.0.0.1:8080/js/showcase.js:2
    fetchQueryByPolicy http://127.0.0.1:8080/js/showcase.js:2
    E http://127.0.0.1:8080/js/showcase.js:2
    fetchQueryObservable http://127.0.0.1:8080/js/showcase.js:2
    fetchQuery http://127.0.0.1:8080/js/showcase.js:2
    query http://127.0.0.1:8080/js/showcase.js:2
    query http://127.0.0.1:8080/js/showcase.js:2
    query http://127.0.0.1:8080/js/showcase.js:2
    modelQuery http://127.0.0.1:8080/js/showcase.js:2
    modelExists http://127.0.0.1:8080/js/showcase.js:2
    loginToModel http://127.0.0.1:8080/js/showcase.js:17
    startAuthAndPolicyModules http://127.0.0.1:8080/js/showcase.js:17
    load http://127.0.0.1:8080/js/showcase.js:17
    loadApplication http://127.0.0.1:8080/js/showcase.js:17
    n http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/?m=ghTF1tNYokM:137
showcase.js:2:569328
[engine] 0.305s MdsReadError: Server response was missing for query 'GetModelPolicies'.
    s http://127.0.0.1:8080/js/showcase.js:2
    s http://127.0.0.1:8080/js/showcase.js:2
    query http://127.0.0.1:8080/js/showcase.js:2
    promise callback*query http://127.0.0.1:8080/js/showcase.js:2
    modelQuery http://127.0.0.1:8080/js/showcase.js:2
    modelExists http://127.0.0.1:8080/js/showcase.js:2
    loginToModel http://127.0.0.1:8080/js/showcase.js:17
    startAuthAndPolicyModules http://127.0.0.1:8080/js/showcase.js:17
    load http://127.0.0.1:8080/js/showcase.js:17
    loadApplication http://127.0.0.1:8080/js/showcase.js:17
    n http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/?m=ghTF1tNYokM:137
showcase.js:17:559812
Uncaught (in promise) MdsReadError: Server response was missing for query 'GetModelPolicies'.
    s http://127.0.0.1:8080/js/showcase.js:2
    s http://127.0.0.1:8080/js/showcase.js:2
    query http://127.0.0.1:8080/js/showcase.js:2
    promise callback*query http://127.0.0.1:8080/js/showcase.js:2
    modelQuery http://127.0.0.1:8080/js/showcase.js:2
    modelExists http://127.0.0.1:8080/js/showcase.js:2
    loginToModel http://127.0.0.1:8080/js/showcase.js:17
    startAuthAndPolicyModules http://127.0.0.1:8080/js/showcase.js:17
    load http://127.0.0.1:8080/js/showcase.js:17
    loadApplication http://127.0.0.1:8080/js/showcase.js:17
    n http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/js/showcase.js:17
    <anonymous> http://127.0.0.1:8080/?m=ghTF1tNYokM:137
showcase.js:2:564823
GET
http://127.0.0.1:8080/images/matterport-logo.svg
NS_BINDING_ABORTED

GET
http://127.0.0.1:8080/images/matterport-logo.svg
NS_BINDING_ABORTED

GET
http://127.0.0.1:8080/images/matterport-logo.svg
[HTTP/1 200 OK 1ms]

XHRPOST
http://127.0.0.1:8080/api/mp/accounts/graph
[HTTP/1 404 File not found 2ms]

GET
http://127.0.0.1:8080/fonts/ibm-plex-sans-700.woff2
[HTTP/1 304 Not Modified 1ms]

XHRPOST
http://127.0.0.1:8080/api/v1/event?ev=error_displayed
[HTTP/1 200 OK 1ms]

XHRPOST
http://127.0.0.1:8080/api/mp/accounts/graph
[HTTP/1 404 File not found 1ms]

XHRPOST
http://127.0.0.1:8080/api/mp/accounts/graph
[HTTP/1 404 File not found 1ms]

XHRPOST
http://127.0.0.1:8080/api/mp/accounts/graph
[HTTP/1 404 File not found 1ms]

XHRPOST
http://127.0.0.1:8080/api/mp/accounts/graph
[HTTP/1 404 File not found 1ms]
```

**Comment by mitchcapper** [comment]
Please provide the server.log  I did a partial download of this and it seemed to work fine.  What browser?

**Comment by HeroponRikiBestest** [comment]
I did those tests on firefox, but it doesn't work on Chrome for me either. 
[server.log](https://github.com/user-attachments/files/17122416/server.log)

**Comment by mitchcapper** [comment]
Are you doing this on localhost?  Unfortunately it looks like your browser was caching things so the server.log doesn't have the initial requests (probably).  Delete the server.log, start it up on a different port (ie 127.0.0.1 9000 then go to http://127.0.0.1:9000 ) it will do the same error but the server.log should get it all.    It is trying to hit an account resource which it doesn't normally access so the questions is why.

**Comment by baransn** [comment]
On MacOS Python 3.12.5 at d65735e8f19016b25c97637e7a842374204562c4

```
python3 matterport-dl.py 'https://my.matterport.com/show/?m=9MxxdwzN2GQ'
python3 matterport-dl.py 'https://my.matterport.com/show/?m=9MxxdwzN2GQ' 127.0.0.1 8082
```

I also see "Oops, model not found"

with the following server log
```
2024-09-24 23:00:06 INFO     Server starting up Running python 3.12.5 on darwin with matterport-dl version: refs/heads/main (d65735e8f19016b25c97637e7a842374204562c4)
2024-09-24 23:00:11 INFO     Redirecting / => /index.modified.html as modified version exists
2024-09-24 23:00:11 INFO     Using our javascript network proxier
2024-09-24 23:00:11 INFO     Redirecting /js/showcase.js => /js/showcase.modified.js as modified version exists
2024-09-24 23:00:11 INFO     Redirecting /?m=9MxxdwzN2GQ => /index.modified.html as modified version exists
2024-09-24 23:00:11 INFO     Using our javascript network proxier
2024-09-24 23:00:11 INFO     Redirecting /js/showcase.js => /js/showcase.modified.js as modified version exists
2024-09-24 23:00:11 INFO     Handling a post request on /api/mp/models/graph?operation=GetModelPolicies: graph of operationName: GetModelPolicies we are handling internally
2024-09-24 23:00:11 WARNING  404 error: /api/mp/accounts/graph may not be downloading everything right
```

**Comment by ASAD-BE18** [comment]
I have also encountered the same issue and upon some investigation, I have found all post requests are getting 415 responses.
```
(Pdb) resp
<Response [415]>
(Pdb) resp.text
'{"code":"request.unsupported.media","message":"Unsupported Media Type"}'
```

**Comment by mitchcapper** [comment]
@ASAD-BE18 thanks for tracking down this problem! I will update our pip file and research this bug.

**Comment by ASAD-BE18** [comment]
With curl-cffi 0.7.2, I have tried posting the request with payload in JSON format, and its response code was 500.

**Comment by mitchcapper** [comment]
Thanks may be worth mentioning in https://github.com/lexiforest/curl_cffi/issues/399 that the json= doesn't seem to fix either (assuming json= does work in 0.7.1)

For everyone else download the latest matterport-dl and run `pip install -U -r requirements.txt`  and it should be fixed.  Also fixed the fact we did not throw an error on JSON post requests failing as we should (which would have made this easier to track down).

**Comment by HeroponRikiBestest** [comment]
Works for me now! Thank you!
